langchain_ydb/vectorstores.py

```diff
@@ -315,20 +317,43 @@ def add_texts(
         metadatas = metadatas if metadatas else [{} for _ in range(len(texts_))]
 
         ydb = self._ydb_lib
-
-        for id, text, metadata in self.pgbar(
-            zip(ids, texts, metadatas),
-            desc="Inserting data...",
-            total=len(ids),
-        ):
-            embedding = self.embedding_function.embed_query(text)
+        
+        # Define struct type with proper member fields
+        document_struct_type = ydb.StructType()
+        document_struct_type.add_member('id', ydb.PrimitiveType.Utf8)
+        document_struct_type.add_member('document', ydb.PrimitiveType.Utf8)
+        document_struct_type.add_member('embedding', ydb.ListType(ydb.PrimitiveType.Float))
+        document_struct_type.add_member('metadata', ydb.PrimitiveType.Json)
+        
+        # Process in batches
+        batch_ranges = range(0, len(texts_), batch_size)
+        for i in self.pgbar(batch_ranges, desc="Processing batches...", total=len(batch_ranges)):
+            batch_texts = texts_[i:i+batch_size]
+            batch_ids = ids[i:i+batch_size]
+            batch_metadatas = metadatas[i:i+batch_size]
+            
+            # Generate embeddings for the batch
+            embeddings = [self.embedding_function.embed_query(text) for text in batch_texts]
```

Comment:
Embeddings have a function like
```
def embed_documents(self, texts: list[str]) -> list[list[float]]:
        """Embed search docs.

        Args:
            texts: List of text to embed.

        Returns:
            List of embeddings.
        """
```

We should use it to batch embeddings as well.